core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+          {% if username %}
+            <li class="dropdown pull-right protractor-test-profile-dropdown">
+              <a class="dropdown-toggle oppia-navbar-dropdown-toggle" data-toggle="dropdown"
+                 ng-mouseover="onMouseoverProfilePictureOrDropdown($event)"
+                 ng-mouseleave="onMouseoutProfilePictureOrDropdown($event)">
+                <div class="oppia-navbar-profile-picture-container" ng-cloak>
+                  {% if profile_picture_data_url %}
+                    <img src="{{profile_picture_data_url}}" class="oppia-navbar-profile-picture img-circle">
+                    <span class="caret" style="margin-top: 10px;"></span>
+                  {% else %}
+                    <i class="material-icons md-40" style="margin-top: -1px;">&#xE853;</i>
+                    <span class="caret" style="margin-top: -26px;"></span>
+                  {% endif %}
+
+                  <div class="oppia-navbar-dashboard-indicator ng-cloak" ng-if="numUnseenNotifications > 0">
+                    <span class="oppia-navbar-dashboard-indicator-text">
+                      <[numUnseenNotifications]>
+                    </span>
+                  </div>
+
+                  {% if is_admin or is_moderator %}
+                    <div class="oppia-navbar-role-indicator">
+                      {% if is_admin %}
+                        <!-- "right: 4px;" is necessary here but not in moderator to prevent 'A' from appearing off-center because 'A' is slightly thinner than 'M' in this font -->
+                        <span class="oppia-navbar-role-text" style="right: 4px;">A</span>
+                      {% elif is_moderator %}
+                        <span class="oppia-navbar-role-text">M</span>
+                      {% endif %}
+                    </div>
+                  {% endif %}
+                </div>
+              </a>
+
+              <ul class="dropdown-menu ng-cloak oppia-navbar-dropdown" role="menu"
+                  ng-mouseover="onMouseoverProfilePictureOrDropdown($event)"
+                  ng-mouseleave="onMouseoutProfilePictureOrDropdown($event)"
+                  ng-show="profileDropdownIsActive">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/profile/{{username}}"
+                     class="protractor-test-profile-link">
+                    <strong>{{username}}</strong>
+                  </a>
+                </li>
+                <hr class="oppia-top-right-menu-item-separator">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/dashboard"
+                     class="protractor-test-dashboard-link">
+                    <span translate="I18N_TOPNAV_DASHBOARD"></span>
+                  </a>
+                </li>
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/notifications_dashboard"
```

Comment:
It's preferable to have a single source of truth for URLs. I would really like to see us get down to just having 1 place which defines all the URLs across the entire site. That being said, it's really a shame that we need to use ``$scope`` in order to move this value to a JS constant. It would be nicer if we could just directly reference constants.